### PR TITLE
Bind `this` In Pilet API Event Functions

### LINKED ISSUE #664: PiletApi's `once` throws when `this` context is `undefined`
# Bug Report

For more information, see the `CONTRIBUTING` guide.

## Prerequisites

- [x] Can you reproduce the problem in a [MWE](https://en.wikipedia.org/wiki/Minimal_working_example)?
- [x] Are you running the latest version?
- [x] Did you perform a search in the issues?

## Environment Details and Version

Piral 1.4.3 on all up-to-date browsers.

## Description

I have code similarly to the following in a Piral app shell React component:

```tsx
const { once, off } = usePiletApi();
// ...

useEffect(() => {
  const handler = () => /* ... */;
  once('my-event', handler);
  return () => off('my-event', handler);
}, [once, off]);
```

This code throws the following error (taken from Firefox): 
grafik

The line originates here:
grafik

## Steps to Reproduce

The issue can easily be reproduced by adding this test case to `api.test.ts`:

```ts
it('can call destructured "once" function without error', () => {
  const events = createListener(undefined);
  const api = initializeApi(moduleMetadata, events);

  // This works ✅
  api.once('my-event', () => {});

  // This fails ❌
  const { once } = api;
  once('my-event', () => {});
});
```

## Expected behavior

`once` can be called in contexts where `this` is `undefined`.

## Actual behavior

`once` throws, as described above.

## Possible Origin/Solution

One solution for this would be the following change in `api.ts`:

```diff
export function initializeApi(target: PiletMetadata, events: EventEmitter): PiletApi {
  return {
    on: events.on,
-   once: events.once,
+   once: events.once.bind(events),
    off: events.off,
    emit: events.emit,
    meta: Object.assign({}, target),
  };
}
```

Not sure if this is the best one or if this can have any side effects? If yes, we could/should prob. do this for all events, not just `once`. I can make a PR if this is fine.

### PR DESCRIPTION
**Pull request by manuelroemer**
# New Pull Request

For more information, see the `CONTRIBUTING` guide.

## Prerequisites

Please make sure you can check the following boxes:

- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [x] All new and existing tests passed

### Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [x] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
- [ ] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes

### Description

Fixes #664.

### Remarks

_None._

### CODE REVIEW

File: src/framework/piral-base/src/api.ts [hunk 1]
```diff
@@ -2,10 +2,10 @@ import type { PiletMetadata, EventEmitter, PiletApi, PiletApiExtender } from './
 
 export function initializeApi(target: PiletMetadata, events: EventEmitter): PiletApi {
   return {
-    on: events.on,
-    once: events.once,
-    off: events.off,
-    emit: events.emit,
+    on: events.on.bind(events),
```
Comment (FlorianRappl) [inline src/framework/piral-base/src/api.ts:5]: Alternatively, solve this already in `events.ts`

Comment (manuelroemer) [reply src/framework/piral-base/src/api.ts:5]: Makes sense, will do!

Comment (manuelroemer) [reply src/framework/piral-base/src/api.ts]: Done. This really is a much better fix for the issue.

**Review by FlorianRappl — APPROVED** [review]
LGTM - thanks!

**Pull request merged**